4exercise2improved.html

```diff
@@ -0,0 +1,60 @@
+<!DOCTYPE html>
+<html>
+
+    <head>
+        <title>Task</title>
+    
+    </head>
+<body>
+
+    <script>
+
+let calculator = {
+    
+    read() {
+        let a;
+        let b;
+        while (isNaN(a) || a === '' || a === null){
```

Comment:
It's better to rewrite the code using an infinite loop to avoid writing the same checks twice.